raysort/sort_utils.py

```diff
@@ -487,3 +487,35 @@ def get_boundaries_auto(
         cfg.num_reducers_per_worker,
         get_boundaries_impl=get_boundaries_impl,
     )
+
+
+def get_median_key(part: np.ndarray) -> float:
+    records = part.reshape((-1, 100))
+    key_bytes = records[:, :8]
+    keys = key_bytes.view(dtype=np.dtype('>u8'))
+    return np.median(keys)
+
+@ray.remote(num_returns=2)
+def split_part(part, pivot, identifier):
+    def id_print(*output):
+        print(":)) split_part: identifier:", identifier, "| output:", output)
+
+
+    if isinstance(part, ray.ObjectRef):        
+        part = ray.get(part)
+    
+    id_print("entered split_part", pivot, part)
+
+    if len(part) == 0:
+        id_print("returning empty list because length is 0: ", len(part))
+        return [], []
+    copy_of_part = part.copy()
```

Comment:
Why copy?

Reply:
because part is an immutable array so we can't call the sort_and_partition function on it.